service/matching/db.go

```diff
@@ -77,6 +78,13 @@ func (db *taskListDB) RangeID() int64 {
 	return db.rangeID
 }
 
+// BacklogCount returns the current backlog size
+func (db *taskListDB) BacklogCount() int64 {
+	db.Lock()
```

Comment:
What about using an atomic counter here rather than locking? it's unlikely to be the source of significant contention, but it should work and it'll avoid any risk to perf